jax/api.py

```diff
@@ -605,12 +647,43 @@ def vmap(fun, in_axes=0, out_axes=0):
            "or a (nested) tuple of those types, got {} and {} respectively.")
     raise TypeError(msg.format(type(in_axes), type(out_axes)))
 
+  def _check_axis_sizes(tree, vals, dims):
+    try:
+      sizes, = {x.shape[d] for x, d in zip(vals, dims) if d is not None}
```

Comment:
Maybe make this a little more explicit and specific:
```python
mapped_axis_sizes = {x.shape[d] for x, d in zip(vals, dims) if d is not None}
try:
  size, = mapped_axis_sizes
except ValueError:
  ...
```

Reply:
Good idea: this means less stuff under the `try` (because what if `x.shape[d]` raises a ValueError?) and also means it's friendlier to interactive debugging inspection. Done.